### ISSUE TITLE
Add joins to sparklingpandas

### ISSUE THREAD

**Issue by michalmonselise** [sparklingpandas/sparklingpandas#121]

**Comment by michalmonselise** [comment]
Could someone assign this issue to me please?

**Comment by jhlch** [comment]
Yes. I'll add you to the org so you can be assigned. Feel free to start working on it. This should be done sometime soon.

**Comment by jhlch** [comment]
Assigned!

**Comment by michalmonselise** [comment]
Thanks :)

I'll be done soon but I had a question:
When we join 2 pandas dataframes and they share a column name (but not the column we are joining on), pandas adds a '_x' and '_y' to the column name to distinguish between the two. Spark does not distinguish between the columns and preserves the original names. How should we handle this? Spark dataframes are immutable so column name changes would require making a copy. 
Should the column names be consistent with spark or pandas and if they are consistent with pandas, would it be ok if I made a copy of the dataframe in order to rename the columns?

**Comment by jhlch** [comment]
Hi Michal,
I think a guiding principal for the sparkling pandas library is to have the
sparkling pandas API match the pandas api as closely as possible. So, I
think this means we should match the pandas style of appending a _x or _y
when column names collide. There is a point of spark clarification that may
make this easier to think about. Spark dataframe's column names are
immutable, but so are dataframes and RDDs themselves. Anytime you apply a
transformation to them, a new pointer to a new data frame is returned. So,
changing column names, or really anything, will result in a new
rdd/dataframe being created. This doesn''t necessarily mean any actual work
happens on the cluster. If we are just changing column names, it should be
a pretty 'cheap' operation that doesn't require any work on the worker
nodes to do. I'd say, just go for it.

Looking forward to your patch! Thanks Michal.

**Comment by michalmonselise** [comment]
I just wanted to update you guys that I'm still working on it. I have a few edge cases to fix that came up during testing.

**Comment by holdenk** [comment]
Awesome, that's exciting :)
